src/praisonai-agents/tests/unit/llm/test_stream_tools_reach_the_model.py

```
@@ -0,0 +1,163 @@
+        for model in ("anthropic/claude-sonnet-4-20250514",
+                      "gemini/gemini-2.0-flash", "ollama/llama3.2"):
+            assert LLM(model=model)._supports_streaming_tools() is False, model
+
+    def test_the_tool_actually_runs(self):
+        llm, state, execute = self._llm_with_two_turns()
+        list(llm.get_response_stream(prompt="weather?", system_prompt="x",
+                                     tools=[get_weather], execute_tool_fn=execute))
+        assert state["ran"] == 1, "the tool call was discarded"
+
+    def test_the_answer_reaches_the_caller(self):
+        llm, state, execute = self._llm_with_two_turns()
```

Ruff (0.16.3)

[warning] Unpacked variable `state` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)